### PR TITLE
Survival functions

### PR DESCRIPTION
**Pull request by kshedden**
I know there are several PR's already in this area but they have been left unfinished for years.  This is a simple implementation of some key missing things, with tests against R.  The quantile estimation and inference is not included in the other work AFAIK.

* Kaplan-Meier estimates of survival functions and plots

* Logrank testing

* Estimation and inference for survival distribution quantiles

### CODE REVIEW

File: statsmodels/duration/survfunc.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    se = d / (n * (n - d)).astype(np.float64)
+    se = np.cumsum(se)
+    se = np.sqrt(se)
+    se *= sp
+
+    return sp, se, time, n, d
+
+
+
+class survfunc_right(object):
```
Comment (josef-pkt) [inline statsmodels/duration/survfunc.py]: pep-8   capitalized class names

File: statsmodels/duration/survfunc.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        Parameters
+        ----------
+        p : float
+            The probability point at which the quantile
+            is determined.
+
+        Returns the estimated quantile.
+        """
+
+        ii = np.flatnonzero(self.surv_prob < 1 - p)
```
Comment (josef-pkt) [inline statsmodels/duration/survfunc.py]: `<=` ?   cdf are usually defined as weak inequality

File: statsmodels/duration/survfunc.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+        if len(ii) == 0:
+            return np.nan
+
+        return self.surv_times[ii[0]]
+
+
+    def quantile_ci(self, p, alpha=0.05):
```
Comment (josef-pkt) [inline statsmodels/duration/survfunc.py]: do we need an option here `method='xxx'`?

**Comment by josef-pkt** [comment]
confidence intervals don't have unit tests

File: statsmodels/duration/survfunc.py (hunk 2 above)
Comment (kshedden) [inline statsmodels/duration/survfunc.py]: I got this from here, where they use a strict inequality:

http://support.sas.com/documentation/cdl/en/statug/63033/HTML/default/viewer.htm#statug_lifetest_sect004.htm

**Comment by kshedden** [comment]
I revised this according to comments from @josef-pkt.  There is now much more testing, against R and SAS.  There are also some new features, including simultaneous confidence intervals for the survival function, more approaches to survival function comparison, and improved documentation.

It runs clear on travis except for the last configuration which seems to fail on everything.

**Comment by josef-pkt** [comment]
I'm merging this also, I didn't see any problems in the code in superficial browsing. The main question is how to combine it with the stalled existing branch.

**Pull request merged**